### PR TITLE
feat(extensions): add PR-comment-formatting extension seam

### PR DESCRIPTION
**Pull request by anderskev**
## Summary

Adds a renderer extension seam so a `daydream_ext` fork can restyle per-finding and summary PR-comment markdown through a new `Registry` renderer slot — without editing `daydream/` and without owning the dedup marker or footer.

Implemented per `.beagle/concepts/pr-comment-formatting-extension/plan.md` (8 TDD tasks, in order).

## What changed

- **Public contract types** (`extensions/api.py`): frozen `CommentFinding`, `FindingRenderContext`, `SummaryFinding`, `SummaryContext`, re-exported from `daydream.extensions`.
- **Registry slot**: `override_renderer` / `renderer` / `renderer_if_registered` / `renderer_names`, mirroring the prompt slot.
- **Default renderers** (`pr_review.py`): `default_render_finding` and `default_render_summary`, registered as builtins (`"finding"`, `"summary"`). Inline, file-level, and summary bodies route through the seam via safe wrappers that fall back to the default and log a warning on any error.
- **`ext validate`**: resolve-checks every registered renderer and reports the count.
- **Docs**: `docs/extensions.md` documents the seam; a drift-guard test keeps the doc in sync with `__all__`.

## Invariants (guarded by tests)

- **Byte-identical default output** — the Task 1 approval snapshots stay green through the Task 4/5 refactors.
- **Host owns the marker and footer** — a renderer never emits `<!-- daydream-finding: ... -->` or `DAYDREAM_FOOTER`; the host injects them around the renderer's inner block.
- **Extension API stays at version 5** (additive change, no floor bump).

## Testing

- New: byte-stable snapshot guard, contract-type frozen/public test, registry slot test, custom-renderer + fallback tests (finding & summary), `ext validate` renderer-count test, end-to-end `$DAYDREAM_EXT_DIR` override test, contract-doc drift guard.
- Also fixed cross-task test hygiene: renderer teardowns now restore the captured registry instead of installing an empty one (was leaking a builtin-less global into the parallel suite), and an E731 lambda assignment replaced with a `def`.
- `make check` (uv lock --check + ruff + mypy + full pytest): green — 3732 passed, 6 skipped.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/extensions.md [hunk 1]
```diff
@@ -433,6 +437,63 @@ fork-selectable schema. A future release that renames this kwarg or replaces it
...
+Renderers return only the inner content. The host owns, and always injects
+around whatever a renderer returns, the parts that dedup and identity depend on:
+
+- the per-finding dedup marker,
+- the `DAYDREAM_FOOTER` trailer,
+- the `---` separators and `<details>` scaffolding,
+- the approval line, and
+- the review `event` decision (approve / comment / request-changes).
+
+A renderer therefore cannot drop the marker or footer, and a custom `"summary"`
```
Comment (anderskev) [inline docs/extensions.md]: 🔵 **Docs 'Host-owned invariants' falsely claims a custom summary renderer omitting body_block still keeps dedup markers; summary markers live inside body_block, so omitting it drops them and causes duplicate re-posting** | severity: `medium` · confidence: `HIGH`

**Severity:** medium

**Confidence:** HIGH

For the summary slot, dedup markers are embedded in SummaryFinding.body_block and only reach output if the renderer emits body_block. reconcile.py parses those markers for cross-run dedup, so a custom summary renderer omitting body_block causes duplicate re-posting. The doc contradicts its own input description and misleads fork authors on a stable public contract. Docs-only, so medium. (Sources: python-records item 1)

<details>
<summary>🔮 Prompt for AI Agents</summary>

```
Verify each finding against the current code and only fix it if needed.

In `docs/extensions.md` around line 482, Docs 'Host-owned invariants' falsely claims a custom summary renderer omitting body_block still keeps dedup markers; summary markers live inside body_block, so omitting it drops them and causes duplicate re-posting: **Severity:** medium
```

</details>

<sub>🧙 Posted by [daydream v0.24.0](https://github.com/existential-birds/daydream)</sub>

File: daydream/extensions/registry.py [hunk 2]
```diff
@@ -160,6 +161,27 @@ def prompt_names(self) -> tuple[str, ...]:
         """Return every registered prompt name in registration order."""
         return tuple(self._prompts)
 
+    # -- renderers --------------------------------------------------------
+
+    def override_renderer(self, name: str, fn: Callable[..., str]) -> None:
+        """Upsert the comment renderer for a named slot."""
```
Comment (anderskev) [inline daydream/extensions/registry.py:167]: 🔵 **override_renderer does not validate callability at registration time, unlike register_tool_supervisor, deferring failure to validation/render time** | severity: `medium` · confidence: `HIGH`

**Severity:** medium

**Confidence:** HIGH

Fail-fast validation on registration is preferable to deferred failure in a validation command or silent fallback at render time. register_tool_supervisor establishes the expected pattern; the inconsistency makes the API contract harder to rely on. (Sources: python-records item 2)

<details>
<summary>🔮 Prompt for AI Agents</summary>

```
Verify each finding against the current code and only fix it if needed.

In `daydream/extensions/registry.py` around line 167, override_renderer does not validate callability at registration time, unlike register_tool_supervisor, deferring failure to validation/render time: **Severity:** medium
```

</details>

<sub>🧙 Posted by [daydream v0.24.0](https://github.com/existential-birds/daydream)</sub>

File: docs/extensions.md [hunk 3]
```diff
@@ -43,13 +43,17 @@ The `daydream.extensions` package exports these contract symbols:
 | `EXTENSION_API_VERSION` | Running extension contract version |
 | `MIN_SUPPORTED_EXTENSION_API_VERSION` | Oldest extension contract version still accepted (range floor) |
 | `BreakLoop` | End the current loop group and continue the flow |
+| `CommentFinding` | Public view of one review finding passed to a `"finding"` renderer |
```
Comment (anderskev) [inline docs/extensions.md:46]: 🔵 **FindingRenderContext placed out of alphabetical order in the Public API Symbols table; ExtensionVersionError should precede FindingRenderContext** | severity: `medium` · confidence: `HIGH`

**Severity:** medium

**Confidence:** HIGH

The convention requires new public types to appear alphabetically in the table. Current order breaks the alphabetical ordering (E before F) that was previously maintained in __all__. (Sources: generic-records item 1)

<details>
<summary>🔮 Prompt for AI Agents</summary>

```
Verify each finding against the current code and only fix it if needed.

In `docs/extensions.md` around line 46, FindingRenderContext placed out of alphabetical order in the Public API Symbols table; ExtensionVersionError should precede FindingRenderContext: **Severity:** medium
```

</details>

<sub>🧙 Posted by [daydream v0.24.0](https://github.com/existential-birds/daydream)</sub>

File: daydream/pr_review.py [hunk 4]
```diff
@@ -717,12 +728,80 @@ def _issue_header(issue: ParsedIssue, *, prefix: str = "", always_bold: bool = F
...
 
+def default_render_finding(finding: CommentFinding, ctx: FindingRenderContext) -> str:
+    """Render the inner human block for one finding (header + body + agent prompt).
+
+    Placement-parameterized (``"inline"``, ``"file_level"``, ``"summary"``) so
+    it reproduces today's inline, file-level, and summary inner text exactly.
+    It never emits :data:`DAYDREAM_FOOTER` or the hidden finding marker — those
+    stay host-owned and are injected by the callers.
+    """
+    issue = ParsedIssue(
```
Comment (anderskev) [inline daydream/pr_review.py]: ⚠️ **default_render_finding rebuilds ParsedIssue from CommentFinding on every call with undocumented field coupling. Consider making _issue_header and _build_agent_prompt accept CommentFinding or a Protocol covering shared fields to avoid reconstruction.** | severity: `high` · confidence: `MEDIUM`

**Severity:** high

**Confidence:** MEDIUM

Field coupling is maintained only by convention with no structural enforcement. A new public field added to CommentFinding would be silently dropped by default_render_finding. Also has mild runtime cost from unnecessary ParsedIssue allocation.

<details>
<summary>🔮 Prompt for AI Agents</summary>

```
Verify each finding against the current code and only fix it if needed.

In `daydream/pr_review.py` around line 739, default_render_finding rebuilds ParsedIssue from CommentFinding on every call with undocumented field coupling. Consider making _issue_header and _build_agent_prompt accept CommentFinding or a Protocol covering shared fields to avoid reconstruction.: **Severity:** high
```

</details>

<sub>🧙 Posted by [daydream v0.24.0](https://github.com/existential-birds/daydream)</sub>

File: daydream/pr_review.py [hunk 5]
```diff
@@ -717,12 +728,80 @@ def _issue_header(issue: ParsedIssue, *, prefix: str = "", always_bold: bool = F
...
+    """Render one finding's inner block through the registered ``"finding"`` renderer.
+
+    Falls back to :func:`default_render_finding` (and warns) when the custom
+    renderer raises or returns a non-``str``/empty result, so a broken fork
+    can never break posting.
+    """
+    cf = _comment_finding(issue)
+    ctx = FindingRenderContext(placement=placement)
+    try:
+        result = get_registry().renderer("finding")(cf, ctx)
```
Comment (anderskev) [inline daydream/pr_review.py]: ⚠️ **Fallback warning text labels the builtin renderer as 'custom' when the builtin itself fails. Update warning message to distinguish between builtin and custom renderers.** | severity: `high` · confidence: `MEDIUM`

**Severity:** high

**Confidence:** MEDIUM

Warning text is misleading when the registered function is the builtin default_render_finding. While unlikely in practice (builtin shouldn't fail under normal operation), the label 'custom' incorrectly describes a builtin fallback scenario.

<details>
<summary>🔮 Prompt for AI Agents</summary>

```
Verify each finding against the current code and only fix it if needed.

In `daydream/pr_review.py` around line 791, Fallback warning text labels the builtin renderer as 'custom' when the builtin itself fails. Update warning message to distinguish between builtin and custom renderers.: **Severity:** high
```

</details>

<sub>🧙 Posted by [daydream v0.24.0](https://github.com/existential-birds/daydream)</sub>

**Review by anderskev — COMMENTED** [review]
**Code Review Summary**

<details>
<summary>📋 Non-inline findings (1)</summary><blockquote>

<details>
<summary>daydream/cli.py (1)</summary><blockquote>

⚠️ **ext validate adds callable() check for renderers but no parallel check for prompts, making validation inconsistently protective. Add a callable() check for prompt names to match renderer validation.** | severity: `high` · confidence: `MEDIUM`

**Severity:** high

**Confidence:** MEDIUM

Both prompt and renderer slots exist for the same purpose (fork-overridable callables) and are documented in the same section. A fork that calls override_prompt with a non-callable passes validation while override_renderer with a non-callable is caught. This gap should be closed for consistency.

<details>
<summary>🔮 Prompt for AI Agents</summary>

```
Verify each finding against the current code and only fix it if needed.

In `daydream/cli.py` around line 1478, ext validate adds callable() check for renderers but no parallel check for prompts, making validation inconsistently protective. Add a callable() check for prompt names to match renderer validation.: **Severity:** high
```

</details>


</blockquote></details>

</blockquote></details>

<details>
<summary>🔮 Prompt for all review comments with AI agents</summary>

```
Fix the 6 review comment(s) posted on this PR.

If the /beagle-core:fetch-pr-feedback skill is available, run:
/beagle-core:fetch-pr-feedback --pr 737

Otherwise, fetch the comments manually:
1. gh api repos/existential-birds/daydream/pulls/737/comments
2. gh api repos/existential-birds/daydream/issues/737/comments

These endpoints return all comments on the PR. Focus on the most
recent review — ignore older review threads that have already been
addressed. For each comment: read the referenced file, verify the
finding against the current code, and fix it if valid. Skip false
positives. Commit all fixes when done.
```

</details>

<details>
<summary>ℹ️ Review info</summary>

- **Model:** mixed — see breakdown
- **Cost:** $6.51
- **Tokens:** 760,184 in (720,577 cached, 95% hit) → 79 out
- **Steps / tool calls:** 15 / 154
- **Duration:** 40m 52s

<details><summary>Per-phase breakdown</summary>

| Phase | Model | Tools | Input (cached) | Output | Cost | Latency |
|---|---|---|---|---|---|---|
| Exploration | claude-sonnet-4-6 | 42 | 162,493 (97%) | 29 | $1.27 | 2m 21s |
| Understand Intent | claude-sonnet-4-6 | 5 | 51,155 (91%) | 1 | $0.26 | 34s |
| Alternatives | claude-opus-4-8 | 9 | 81,035 (100%) | 9 | $1.13 | 3m 15s |
| Deep Review | mixed | 95 | 352,309 (97%) | 37 | $3.72 | 34m 39s |
| Parse Feedback | claude-haiku-4-5-20251001 | 3 | 113,192 (83%) | 3 | $0.13 | 28s |

</details>

<sub>Generated by daydream v0.24.0</sub>

- **Severity:** 3 high, 3 medium
- **Confidence:** 3 HIGH, 3 MEDIUM

</details>

<sub>🧙 Posted by [daydream v0.24.0](https://github.com/existential-birds/daydream)</sub>
